packages/components/src/components/breadcrumb/breadcrumb-item.tsx

```diff
@@ -0,0 +1,79 @@
+import { Component, h, Element, State, Listen } from '@stencil/core';
+
+@Component({
+  tag: 'ifx-breadcrumb-item',
+  styleUrl: 'breadcrumb-item.scss',
+  shadow: true,
+})
+export class BreadcrumbItem {
+
+  @State() isLastItem: boolean = false;
+  @Element() el;
+
+  @Listen('mousedown', { target: 'document' })
+  handleOutsideClick(event: MouseEvent) {
+    const path = event.composedPath();
+    if (!path.includes(this.el)) {
+      this.closeDropdownMenu();
+    }
+  }
+
+  getDropdownMenu() {
+    const dropdownMenu = this.el.shadowRoot.querySelector('.dropdown-menu');
+    return dropdownMenu
+  }
+
+  getMenuIconWrapper() { 
+    const menuIconWrapper = this.el.querySelector('ifx-breadcrumb-item-label')
+    const container = menuIconWrapper.shadowRoot.querySelector('.breadcrumb-item-label-container')
+    const menuWrapper = container.querySelector('.menu-icon-wrapper')
+    return menuWrapper
+  }
+
+  handleClassList(el, type, className) {
+    el.classList[type](className)
+  }
+
+  closeDropdownMenu() {
+    const dropdownMenu = this.getDropdownMenu()
+    const menuWrapper = this.getMenuIconWrapper()
+    this.handleClassList(dropdownMenu, 'remove', 'open')
+    this.handleClassList(menuWrapper, 'remove', 'show')
+  }
+
+  toggleDropdownMenu() { 
+    const dropdownMenu = this.getDropdownMenu()
+    const menuWrapper = this.getMenuIconWrapper()
+    this.handleClassList(dropdownMenu, 'toggle', 'open')
+    this.handleClassList(menuWrapper, 'toggle', 'show')
+  }
+
+  componentWillLoad() { 
+    const breadcrumbItems = this.el.closest('ifx-breadcrumb').querySelectorAll('ifx-breadcrumb-item')
+    if(this.el === breadcrumbItems[breadcrumbItems.length-1]) { 
+      this.isLastItem = true;
+    } else this.isLastItem = false;
+  }
+
+  componentDidLoad() { 
+    const dropdownMenu = this.el.querySelector('ifx-dropdown-menu')
+    if(!dropdownMenu) { 
+      const iconMenuWrapper = this.getMenuIconWrapper();
+      this.handleClassList(iconMenuWrapper, 'toggle', 'hide')
+    }
+  }
+
+  render() {
+    return (
+      <li class="breadcrumb-parent" aria-current="" onClick={this.toggleDropdownMenu.bind(this)}>
```

Comment:
Improper use of bind: The use of .bind(this) in the JSX could be avoided by changing your class methods to arrow functions. This ensures that this is automatically bound to the class instance.

Reply:
Why is this an improper use of bind? To my knowledge, this is a completely common and proper way of binding the element. What makes it improper?

Reply:
You can use an inline function for the same approach. This is mainly personal preferences that comes from these points. But on todays standard the performance impact is almost not existent. Just to be clear: you do not need .bind here. It doesn't hurt to remove it, right? ;)

Performance issues: Every time a component renders, it will create a new function if you use .bind() in the render method. This can lead to unnecessary memory usage and potential performance degradation, especially for components that re-render frequently.

Prevents optimizations: JavaScript engines and frontend frameworks often have optimization techniques to prevent unnecessary updates and renders. However, when you bind a function in the render method, you're creating a new function every time, and the optimization techniques can't recognize that it's conceptually the same function, thus leading to unnecessary re-renders or computations.

Non-idiomatic: It's considered a better practice to bind event handlers in the constructor of the component or use class properties to avoid creating a new function every render. This makes the code more idiomatic and easier to understand.

Reply:
I've changed it to an arrow function, but you do need to use bind(this) if you do not use an arrow function, because you need to bind the 'this' to the component, otherwise, 'this' inside the function will be undefined.